### PR TITLE
Adds a new type of botany experiment that tests plants.

### PR DESCRIPTION
**Pull request by ArcaneMusic**
## About The Pull Request

Adds two new experiments, the plant trait experiment and the wild mutation plant experiment which boosts experimental surgery and the botany node respectively.

The wild mutation experiment is rather simple, and actually helps teach players on how to use wild experiments in order to get free mutations out of plants without having to fiddle with instability for as long. Just go and use the deconstructive scanner (which is a public machine on each map) with a plant that was mutated with this method, and the botany node is now cheaper.

The plant trait experiment looks for a few plants with a collection of random traits, all selected from some of the harder traits to aquire. You might need to acquire some plants with googly eyes, or shadowshrooms, or perhaps even prickly adhesion. Regardless, any plant with one of the requested traits will be all it takes, and it boosts the surgery node.

## Why It's Good For The Game

This gives botany another way to interact with the crew, makes their own stuff a little cheaper if science is being cheapstakes, and gives a neat framework to enable botany to hit unique milestones within the round to benefit the rest of the crew. More per-round engagement and inter-departmental cooperation creates a more lively round. Plus it gives the new and old botany powergamers something to do that isn't feeding toxic soybeans into the tapwater.

## Changelog
:cl:
add: There are now two experiments that botany can perform that boosts botany tech and experimental surgery.
/:cl:

### CODE REVIEW

**Comment by bobbahbrown** [comment]
@Qustinnus worth taking a peek when you get a chance

File: code/modules/experisci/experiment/types/scanning_plants.dm [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+/datum/experiment/scanning/random/plants/serialize_progress_stage(atom/target, list/seen_instances)
+	return EXPERIMENT_PROG_INT("Scan samples of \a harvested plant.", \
+		traits & EXPERIMENT_TRAIT_DESTRUCTIVE ? scanned[target] : seen_instances.len, required_atoms[target])
+
+/datum/experiment/scanning/random/plants/traits/final_contributing_index_checks(atom/target, typepath)
+	if(!istype(target, /obj/item/food/grown))
+		return FALSE
+	var/obj/item/food/grown/crop = target
+	if(possible_plant_genes.len)
+		return ..() && crop?.seed.get_gene(required_genes[typepath])
```
Comment (Qustinnus) [inline code/modules/experisci/experiment/types/scanning_plants.dm:29]: crop cant be nul lhere

File: code/modules/research/techweb/all_nodes.dm [hunk 2]
```diff
@@ -117,6 +117,7 @@
 	prereq_ids = list("adv_surgery")
 	design_ids = list("surgery_pacify","surgery_vein_thread","surgery_muscled_veins","surgery_nerve_splice","surgery_nerve_ground","surgery_ligament_hook","surgery_ligament_reinforcement","surgery_cortex_imprint","surgery_cortex_folding","surgery_viral_bond", "surgery_heal_combo_upgrade")
 	research_costs = list(TECHWEB_POINT_TYPE_GENERIC = 5000)
+	discount_experiments = list(/datum/experiment/scanning/random/plants/traits = 2500)
```
Comment (Qustinnus) [inline code/modules/research/techweb/all_nodes.dm:120]: personal rpeference is that you increase the cost of the node as well so its more likely people will do the experiment

**Review by ghost — CHANGES_REQUESTED** [review]

File: code/modules/research/techweb/all_nodes.dm (hunk 2 above)
Comment (Qustinnus) [reply code/modules/research/techweb/all_nodes.dm:120]: (and then make it so that the experiment cuts out a bigger portion of that cost)

**Review by ghost — APPROVED** [review]

File: code/__DEFINES/traits.dm [hunk 3]
```diff
@@ -328,6 +328,7 @@ Remember to update _globalvars/traits.dm if you're adding/removing/renaming trai
 #define TRAIT_NEEDS_TWO_HANDS "needstwohands" //The items needs two hands to be carried
 #define TRAIT_FISH_SAFE_STORAGE "fish_case" //Fish in this won't die
 #define TRAIT_FISH_CASE_COMPATIBILE "fish_case_compatibile" //Stuff that can go inside fish cases
+#define TRAIT_PLANT_WILDMUTATE "wildmutation" //Plants that were mutated as a result of passive instability, not a mutation threshold.
```
Comment (Mothblocks) [inline code/__DEFINES/traits.dm:331]:
```suggestion
/// Plants that were mutated as a result of passive instability, not a mutation threshold.
#define TRAIT_PLANT_WILDMUTATE "wildmutation"
```

Doc comments.

**Pull request merged**